test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go

```diff
@@ -1162,56 +1166,67 @@ var _ = Describe("[rfe_id:28761][performance] Updating parameters in performance
 			By("checking if ContainerRuntimeConfig object already exists")
 			ctrcfg, err = getContainerRuntimeConfigFrom(context.TODO(), profile, mcp)
 			Expect(err).ToNot(HaveOccurred(), "failed to get ContainerRuntimeConfig from profile %q mcp %q", profile.Name, mcp.Name)
+			if ctrcfg != nil {
+				Skip(fmt.Sprintf("ContainerRuntimeConfig %s exist in the cluster and not expected", ctrcfg.Name))
+			}
+			testpodTemplate = pods.GetTestPod()
+			testpodTemplate.Namespace = testutils.NamespaceTesting
+			runtimeClass := components.GetComponentName(profile.Name, components.ComponentNamePrefix)
+			testpodTemplate.Spec.RuntimeClassName = &runtimeClass
 		})
-
-		When("is not given", func() {
-			It("should run high-performance runtimes class with runc as container-runtime", func() {
-				if ctrcfg != nil {
-					Skip("runc is not the default runtime configuration")
-				}
-				cmd := []string{"cat", "/rootfs/etc/crio/crio.conf.d/99-runtimes.conf"}
-				for i := 0; i < len(workerRTNodes); i++ {
-					output, err := nodes.ExecCommand(context.TODO(), &workerRTNodes[i], cmd)
-					Expect(err).ToNot(HaveOccurred(), "cannot get 99-runtimes.conf from %q", workerRTNodes[i].Name)
-					out := testutils.ToString(output)
-					By(fmt.Sprintf("checking node: %q", workerRTNodes[i].Name))
-					Expect(out).To(ContainSubstring("/bin/runc"))
-					Expect(out).To(ContainSubstring("/run/runc"))
-				}
-			})
-		})
-		When("updates the default runtime to crun", func() {
-			It("should run high-performance runtimes class with crun as container-runtime", func() {
-				if ctrcfg == nil {
-					testlog.Infof("ContainerRuntimeConfig not exist")
+		DescribeTable("verifies container runtime behavior",
+			func(withCTRCfg bool) {
+				var expectedRuntime string
+				if withCTRCfg {
 					ctrcfg = newContainerRuntimeConfig(ContainerRuntimeConfigName, profile, mcp)
 					By(fmt.Sprintf("creating ContainerRuntimeConfig %q", ctrcfg.Name))
-					Expect(testclient.Client.Create(context.TODO(), ctrcfg)).ToNot(HaveOccurred(), "failed to create ctrcfg %#v", ctrcfg)
+					Expect(testclient.ControlPlaneClient.Create(context.TODO(), ctrcfg)).ToNot(HaveOccurred(), "failed to create ctrcfg %#v", ctrcfg)
```

Comment:
It won't work on HyperShift.
You'll have to create the object on the `clusters` Namespace and then attaches it to the nodePool

Reply:
Yes you are right. anyway, this test has the openshift label and currently not running on hypershift lanes.
I figured that out later. 
I can modify this to use the `testclient.client` as before if you would like

Reply:
@SargunNarula I saw your PR. The changes for supporting hypershift belong to a separate (your PR).
Therefore, I have removed my client adjustments here, please ensure you are doing them in your PR.